editor/src/clj/editor/lua.clj

```diff
@@ -181,6 +181,19 @@
 (def defold-docs (defold-documentation))
 (def preinstalled-modules (into #{} (remove #{""} (keys defold-docs))))
 
+(def base-globals
+  (into #{}
+        (map #(.getName ^ScriptDoc$Element %))
+        (get (load-sdoc (sdoc-path "base")) "")))
+
+(def defined-globals
+  (into #{}
+        (comp
+          (remove #(= :message (:type %)))
+          (map #(-> % :name (string/split #":" 2) first))
+          (remove (conj base-globals "")))
+        (get defold-docs "")))
```

Comment:
This appears to include some stuff from the Lua standard library, such as the `string` and `table` modules. Is that intentional?

I think this should report the same set of modules as we have in our API documentation, but currently this doesn't match. See here: https://defold.com/ref/stable/go/

Also, is is enough to just declare the modules like `go`? Don't we want it to declare its functions, like `go.set_position`, for example?

Reply:
Hmm good point, I wonder why we are getting them. 

Regarding functions — we don't need to define them, because luacheck does not try to validate that a function in a referenced module exists, only if a module itself exists. Surprising, yes, but it seems almost every lua linter works that way.

Reply:
It is actually possibly to also define available functions on a module. Maybe something to add in a future update?

Reply:
Yes, I created an issue for that https://github.com/defold/lua-language-server/issues/8